library/alignment.h

```diff
@@ -110,11 +202,25 @@ inline uint64_t mbedtls_get_unaligned_uint64(const void *p)
  * \param   p pointer to 8 bytes of data
  * \param   x data to write
  */
-inline void mbedtls_put_unaligned_uint64(void *p, uint64_t x)
+#if defined(__IAR_SYSTEMS_ICC__)
+#pragma inline = forced
+#elif defined(__GNUC__)
+__attribute__((always_inline))
+#endif
+static inline void mbedtls_put_unaligned_uint64(void *p, uint64_t x)
 {
+#if defined(UINT_UNALIGNED)
+    mbedtls_uint64_unaligned_t *p64 = (mbedtls_uint64_unaligned_t *) p;
+    *p64 = x;
+#else
     memcpy(p, &x, sizeof(x));
+#endif
 }
 
+#if defined(UINT_UNALIGNED) && defined(__IAR_SYSTEMS_ICC__)
```

Comment:
This isn't quite the same condition under which `#pragma language=` is given. Although it's fine for now, I wonder about maintainability. It looks like IAR supports `_Pragma()`, would allow us to do

```
#define MBEDTLS_POP_LANGUAGE_PRAGMA

#if defined(__IAR_SYSTEMS_ICC__) && \
...
#undef MBEDTLS_POP_LANGUAGE_PRAGMA
#define MBEDTLS_POP_LANGUAGE_PRAGMA _Pragma("language=restore")
:
#endif
```

Then here use

```
MBEDTLS_POP_LANGUAGE_PRAGMA
```

Reply:
I've done this slightly differently as your suggestion made code_style.py quite confused.